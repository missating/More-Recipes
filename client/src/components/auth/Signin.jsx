import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

// validations
import signinValidator from '../../validation/signinValidator';

// actions
import fetchUserSignin from '../../actions/userSignIn';


/**
 * @description Creates Signin component
 *
 * @class Signin
 *
 * @extends {React.Component}
 */
export class Signin extends React.Component {
  /**
   * @description Creates an instance of Signin.
   *
   * @constructor
   *
   * @param {any} props
   *
   * @memberof Signin
   *
   * @returns {void}
   */
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      errors: {}
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  /**
   * @description set state
   *
   * @param {any} nextProps
   *
   * @memberof Signin
   *
   * @returns {void}
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.auth.isAuthenticated &&
      nextProps.auth.errorMessage.length === 0) {
      $('#signin-modal').modal('hide');
      this.setState({
        email: '',
        password: '',
      });
    }
  }


  /**
   * @description Bind the value of the inputs to state
   *
   * @method onChange
   *
   * @param {any} event
   *
   * @memberof Signin
   *
   * @returns {void}
   */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  /**
   * @description Submit the form
   *
   * @method onSubmit
   *
   * @param {object} event
   *
   * @memberof Signin
   *
   * @returns {void}
   */
  onSubmit(event) {
    event.preventDefault();
    const isValid = this.isValid();
    if (isValid) {
      this.props.signinUser(this.state);
    }
  }

  /**
    * @description Validates user's data before making post request
    *
    * @method isValid
    *
    * @memberof Signin
    *
    * @returns {boolean} true or false
    */
  isValid() {
    const { isValid, errors } = signinValidator(this.state);
    if (!isValid) {
      this.setState({ errors });
    } else {
      this.setState({ errors: {} });
      return isValid;
    }
  }


  /**
   * @description Renders react component
   *
   * @method render
   *
   * @memberof Signin
   *
   * @returns {void}
   *
   */
  render() {
    const { errors } = this.state;

    let signinError;
    if (this.props.auth.errorMessage) {
      signinError = (
        <span className="help-block text-danger">
          {this.props.auth.errorMessage}
        </span>
      );
    }
    return (
      <div>
        <div
          id="signin-modal"
          className="modal fade"
        >
          <div className="modal-dialog modal-dialog-centered modal-sm">
            <div className="modal-content">
              <button
                type="button"
                className="close"
                data-dismiss="modal"
                aria-label="Close"
              >
                <span
                  aria-hidden="true"
                  style={{
                    float: 'right',
                    cursor: 'pointer'
                  }}
                >
                  &times;
                </span>
              </button>
              <div className="modal-body">

                {signinError}

                <div className="row">
                  <div className="col-sm-12">
                    <form onSubmit={this.onSubmit}>

                      <div className="form-group">
                        <input
                          type="email"
                          placeholder="Email"
                          name="email"
                          value={this.state.email}
                          onChange={this.onChange}
                          className="form-control signin-input"
                        />
                      </div>

                      <div className="form-group">
                        <input
                          type="password"
                          placeholder="Password"
                          id="signinPassword"
                          name="password"
                          value={this.state.password}
                          onChange={this.onChange}
                          className="form-control signin-input"
                        />
                      </div>
                      {
                        errors.email &&
                        <span className="help-block text-danger">
                          {errors.email}
                        </span>
                      }
                      <br />
                      {
                        errors.password &&
                        <span className="help-block text-danger">
                          {errors.password}
                        </span>
                      }
                      <div />

                      <button
                        className="btn btn-secondary"
                        type="submit"
                      >
                        Submit
                      </button>
                    </form>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

Signin.propTypes = {
  auth: PropTypes.shape({
    isAuthenticated: PropTypes.bool.isRequired,
    user: PropTypes.shape({
      fullname: PropTypes.string,
      username: PropTypes.string,
      email: PropTypes.string,
      id: PropTypes.number
    }),
    errorMessage: PropTypes.string
  }).isRequired,
  signinUser: PropTypes.func.isRequired
};

const mapDispatchToProps = dispatch => ({
  signinUser: (userDetails) => {
    dispatch(fetchUserSignin(userDetails));
  }
});

const mapStateToProps = state => ({
  auth: state.auth
});

const ConnectedComponent =
  connect(mapStateToProps, mapDispatchToProps)(Signin);
export { ConnectedComponent as ConnectedSignin };
