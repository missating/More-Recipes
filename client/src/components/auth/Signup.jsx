import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

// validations
import signupValidator from '../../validation/signupValidator';

// actions
import fetchUserSignup from '../../actions/signup';


/**
 * @description Creates Signup component
 *
 * @class Signup
 *
 * @extends {React.Component}
 */
export class Signup extends React.Component {
  /**
   * @description Creates an instance of Signup.
   *
   * @constructor
   *
   * @param {any} props
   *
   * @memberof Signup
   *
   * @returns {void}
   */
  constructor(props) {
    super(props);
    this.state = {
      fullname: '',
      username: '',
      email: '',
      password: '',
      confirmPassword: '',
      errors: {}
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  /**
   * @description Sets state
   *
   * @param {any} nextProps
   *
   * @memberof Signup
   *
   * @returns {void}
   *
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.auth.isAuthenticated &&
      nextProps.auth.errorMessage.length === 0) {
      $('#signup-modal').modal('hide');
      this.setState({
        fullname: '',
        username: '',
        email: '',
        password: '',
        confirmPassword: '',
      });
    }
  }

  /**
   * @description Bind the value of the inputs to state
   *
   * @method onChange
   *
   * @memberof Signup
   *
   * @param {any} event
   *
   * @returns {void}
   */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  /**
   * @description Submit the form
   *
   * @method onSubmit
   *
   * @param {object} event
   *
   * @memberof Signup
   *
   * @returns {void}
   */
  onSubmit(event) {
    event.preventDefault();
    const isValid = this.isValid();
    if (isValid) {
      this.props.signupUser(this.state);
    }
  }

  /**
   * @description Validates user's data before making post request
   *
   * @method isValid
   *
   * @memberof Signup
   *
   * @returns {boolean} true or false
   */
  isValid() {
    const { isValid, errors } = signupValidator(this.state);
    if (!isValid) {
      this.setState({ errors });
    } else {
      this.setState({ errors: {} });
      return isValid;
    }
  }

  /**
   * @description Renders react component
   *
   * @method render
   *
   * @memberof Signup
   *
   * @returns {void}
   *
   */
  render() {
    const { errors } = this.state;

    let signupError;
    if (this.props.auth.errorMessage) {
      signupError = (
        <span className="help-block text-danger">
          {this.props.auth.errorMessage}
        </span>
      );
    }
    return (
      <div>
        <div
          id="signup-modal"
          className="modal fade"
        >
          <div className="modal-dialog modal-dialog-centered modal-sm">
            <div className="modal-content">
              <button
                type="button"
                className="close"
                data-dismiss="modal"
                aria-label="Close"
              >
                <span
                  aria-hidden="true"
                  style={{
                    float: 'right',
                    cursor: 'pointer'
                  }}
                >
                  &times;
                </span>
              </button>
              <div className="modal-body">

                {signupError}

                <div className="row">
                  <div className="col-sm-12">
                    <form onSubmit={this.onSubmit}>

                      <div className="form-group">
                        <input
                          type="text"
                          name="fullname"
                          className="form-control"
                          placeholder="Full Name"
                          value={this.state.fullname}
                          onChange={this.onChange}
                        />
                        {
                          errors.fullname &&
                          <span className="help-block text-danger">
                            {errors.fullname}
                          </span>
                        }
                      </div>

                      <div className="form-group">
                        <input
                          type="text"
                          name="username"
                          className="form-control"
                          placeholder="User Name"
                          value={this.state.username}
                          onChange={this.onChange}
                        />
                        {
                          errors.username &&
                          <span className="help-block text-danger">
                            {errors.username}
                          </span>
                        }
                      </div>

                      <div className="form-group">
                        <input
                          type="email"
                          name="email"
                          className="form-control"
                          placeholder="Email Address"
                          value={this.state.email}
                          onChange={this.onChange}
                        />
                        {
                          errors.email &&
                          <span className="help-block text-danger">
                            {errors.email}
                          </span>
                        }
                      </div>

                      <div className="form-group">
                        <input
                          type="password"
                          name="password"
                          className="form-control"
                          placeholder="Password"
                          value={this.state.password}
                          onChange={this.onChange}
                        />
                        {
                          errors.password &&
                          <span className="help-block text-danger">
                            {errors.password}
                          </span>
                        }
                      </div>


                      <div className="form-group">
                        <input
                          type="password"
                          id="signupPassword"
                          name="confirmPassword"
                          className="form-control"
                          placeholder="Confirm Password"
                          value={this.state.confirmPassword}
                          onChange={this.onChange}
                        />
                        {
                          errors.confirmPassword &&
                          <span className="help-block text-danger">
                            {errors.confirmPassword}
                          </span>
                        }
                      </div>
                      <button
                        className="btn btn-secondary"
                        type="submit"
                      >
                        Submit
                      </button>
                    </form>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}


Signup.propTypes = {
  signupUser: PropTypes.func.isRequired,
  auth: PropTypes.shape({
    isAuthenticated: PropTypes.bool.isRequired,
    user: PropTypes.shape({
      fullname: PropTypes.string,
      username: PropTypes.string,
      email: PropTypes.string,
      id: PropTypes.number
    }),
    errorMessage: PropTypes.string
  }).isRequired,
};

const mapDispatchToProps = dispatch => ({
  signupUser: (userDetails) => {
    dispatch(fetchUserSignup(userDetails));
  }
});

const mapStateToProps = state => ({
  auth: state.auth
});

const ConnectedComponent =
  connect(mapStateToProps, mapDispatchToProps)(Signup);
export { ConnectedComponent as ConnectedSignup };
