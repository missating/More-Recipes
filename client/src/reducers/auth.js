import {
  RECEIVE_AUTH,
  AUTH_ERROR,
  SIGN_IN_USER,
  SIGN_OUT
} from '../actions/actionTypes';

const initialState = {
  isAuthenticated: !!localStorage.getItem('token'),
  user: JSON.parse(localStorage.getItem('user'))
};

const auth = (state = initialState, action) => {
  switch (action.type) {
  case RECEIVE_AUTH:
    return {
      ...state,
      isAuthenticated: true,
      errorMessage: '',
      user: action.user,
      token: action.token
    };
  case AUTH_ERROR:
    return {
      ...state,
      isAuthenticated: false,
      errorMessage: action.message
    };
  case SIGN_IN_USER:
    return {
      ...state,
      isAuthenticated: true,
      errorMessage: '',
      user: action.user,
      token: action.token
    };
  case SIGN_OUT:
    return {
      ...state,
      isAuthenticated: false,
      user: {},
      token: ''
    };
  default:
    return state;
  }
};

export default auth;
