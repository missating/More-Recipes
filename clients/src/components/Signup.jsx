import React from 'react';
import propTypes from 'prop-types';
import signupValidator from '../validation/signupValidator';
import userSignupRequest from '../actions/signupActions';
import { browserHistory } from 'react-router';

import { connect } from 'react-redux';

import '../css/style.css';

class Signup extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
        fullname: 'jjajaja  ',
        username: 'sjsjjss',
        email: 'jjsjs@ssks.com',
        password: '123456',
        confirmPassword: '123456',
        errors: {},
        serverError: null,
        isLoading: false
    }
		this.onChange = this.onChange.bind(this);
		this.onSubmit = this.onSubmit.bind(this);
  }

  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  isValid() {
		const { isValid , errors } = signupValidator(this.state);
		if (!isValid) {
			this.setState({ errors });
		}
		return isValid;
	}
  

  onSubmit(event) {
    event.preventDefault();
    this.props.dispatch(userSignupRequest(this.state));
    // if (this.isValid()) {
		// 	this.setState({ errors: {}, isLoading: true });
		// 	this.props.userSignupRequest(this.state).then(response => {
    //     console.log(response);
    //     $("#myModal").modal('hide');
    //     this.props.history.push('/recipes');
    //   }).catch(errors => {
    //     console.log(errors);
    //     this.setState({
    //       serverError: errors.data.message
    //     });
    //   });
		// }
  }

	
  render() {
    const { errors } = this.state;
    let serverErrorBag;
    if (this.state.serverError) {
      serverErrorBag = (
        <span className="help-block">{this.state.serverError}</span>
      );
    }
    return (
      <div> 
        <div id="myModal" className="modal fade" tabIndex="-1" role="dialog" aria-hidden="true">
            <div className="modal-dialog modal-lg">
              <div className="modal-content">
                  <div className="modal-body">

                      {serverErrorBag}

                      <div className="row">
                          <div className="col-sm-12">
                          <form className="form-horizontal" onSubmit={this.onSubmit}>
                            
                          <div className="form-group">
                                <input type="text" name="fullname" className="form-control" placeholder='Full Name'
                                value={this.state.fullname} 
                                onChange={this.onChange} />
                                 {errors.fullname && <span className="help-block">{errors.fullname}</span>}
                          </div>
                      
                          <div className="form-group">
                                <input type="text" name="username" className="form-control" placeholder='User Name'
                                value={this.state.username} 
                                onChange={this.onChange}  />
                                {errors.username && <span className="help-block">{errors.username}</span>}
                          </div>
                        
                          <div className="form-group">
                                <input type="email" name="email" className="form-control" placeholder='Email Address'
                               value={this.state.email} 
                               onChange={this.onChange} />  
                             {errors.email && <span className="help-block">{errors.email}</span>}
                          </div>
                       
                          <div className="form-group">
                                <input type="password" name="password" className="form-control" placeholder='Password'
                                value={this.state.password} 
                                onChange={this.onChange} /> 
                                 {errors.password && <span className="help-block">{errors.password}</span>}
                          </div>
                        
        
                          <div className="form-group">
                                <input type="password" name="confirmPassword" className="form-control" placeholder='Confirm Password' 
                                value={this.state.confirmPassword} 
                                onChange={this.onChange} />
                                {errors.confirmPassword && <span className="help-block">{errors.confirmPassword}</span>}
                          </div>
                        <br/>

                          <button
                              className="btn btn-primary btn-sm" type="submit"> Submit
                          </button>
                        </form>
                        </div>
                      </div>
                    </div>
                      <div className="modal-footer">
                              <button disabled={this.state.isLoading} className="btn btn-danger btn-sm" data-dismiss="modal">close</button>
                          </div>
                      </div>
                        </div>
                    </div>
                </div>
        )
    }
}

Signup.propTypes = {
	userSignupRequest: propTypes.func.isRequired
}

export default connect(null)(Signup);
