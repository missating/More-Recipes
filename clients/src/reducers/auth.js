const initialState = {
  isAuthenticated: !!localStorage.getItem('token'),
};

const auth = (state = initialState, action) => {
  switch (action.type) {
    case 'COOK_RICE':
      console.log(action);
      return state;
    default:
      return state;
  }
};

export default auth;
