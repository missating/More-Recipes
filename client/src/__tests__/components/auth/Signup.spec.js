import React from 'react';
import toJSON from 'enzyme-to-json';
import sinon from 'sinon';
import { Signup } from '../../../components/auth/Signup';

describe('Sign up component', () => {
  const props = {
    signupUser: jest.fn(() => Promise.resolve()),
    auth: {
      isAuthenticated: true,
      user: {
        fullname: 'test',
        username: 'test',
        email: 'test@test.com',
        id: 1
      },
      errorMessage: ''
    }
  };

  it('Should render', () => {
    const wrapper = shallow(<Signup
      {...props}
    />);
    expect(wrapper.find('form').exists).toBeTruthy();
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it(
    'calls onChange method when the value of the new user input changes',
    () => {
      const spy = sinon.spy(Signup.prototype, 'onChange');
      const wrapper = shallow(<Signup
        {...props}
      />);
      wrapper.find('#signupPassword').simulate('change', {
        target: { name: 'confirmPassword', value: '123456' }
      });
      expect(spy.called).toBeTruthy();
    }
  );


  it('Should submit form when submit button is clicked', () => {
    const spy = sinon.spy(Signup.prototype, 'onSubmit');
    const wrapper = shallow(<Signup
      {...props}
    />);
    wrapper.find('form').simulate('submit', { preventDefault: jest.fn() });
    expect(spy.called).toBeTruthy();
  });


  it('Should add user when form submit with all user details', () => {
    const wrapper = shallow(<Signup
      {...props}
    />);
    wrapper.instance().setState({
      fullname: 'test',
      username: 'test',
      email: 'test@test.com',
      password: '123456',
      confirmPassword: '123456',
      errors: {}
    });
    wrapper.update();
    wrapper.find('form').simulate('submit', { preventDefault: jest.fn() });
    expect(wrapper.instance().state).toEqual({
      fullname: 'test',
      username: 'test',
      email: 'test@test.com',
      password: '123456',
      confirmPassword: '123456',
      errors: {}
    });
  });


  it('Should be called when component is passed in new props', () => {
    const wrapper = shallow(<Signup
      {...props}
    />);
    wrapper.instance().componentWillReceiveProps(props);
    expect(wrapper.instance().props.auth.isAuthenticated)
      .toEqual(props.auth.isAuthenticated);
  });
});

