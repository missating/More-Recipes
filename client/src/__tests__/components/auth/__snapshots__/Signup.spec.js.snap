// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign up component Should render 1`] = `
<div>
  <div
    className="modal fade"
    id="signup-modal"
  >
    <div
      className="modal-dialog modal-dialog-centered modal-sm"
    >
      <div
        className="modal-content"
      >
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
            style={
              Object {
                "cursor": "pointer",
                "float": "right",
              }
            }
          >
            Ã—
          </span>
        </button>
        <div
          className="modal-body"
        >
          <div
            className="row"
          >
            <div
              className="col-sm-12"
            >
              <form
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <input
                    className="form-control"
                    name="fullname"
                    onChange={[Function]}
                    placeholder="Full Name"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <input
                    className="form-control"
                    name="username"
                    onChange={[Function]}
                    placeholder="User Name"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <input
                    className="form-control"
                    name="email"
                    onChange={[Function]}
                    placeholder="Email Address"
                    type="email"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <input
                    className="form-control"
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <input
                    className="form-control"
                    id="signupPassword"
                    name="confirmPassword"
                    onChange={[Function]}
                    placeholder="Confirm Password"
                    type="password"
                    value=""
                  />
                </div>
                <button
                  className="btn btn-secondary"
                  type="submit"
                >
                  Submit
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
